components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,933 @@
+    call register_variable("example_pio_structured_v2.nc","foo","answer to space and time",3,(/ "x", "y", "time" /), PIO_double,"xyt-real")
+    call register_variable("example_pio_structured_v2.nc","bar_flip","answer to space and time",3,(/ "y", "x", "time" /), PIO_double,"yxt-real")
+    call register_variable("example_pio_structured_v2.nc","foo_flip","answer to space and time",3,(/ "y", "x", "time" /), PIO_double,"yxt-real")
+
+    call get_pio_atm_File(trim("example_pio_structured_v2.nc"),current_atm_file,found)
+    ierr = PIO_enddef(current_atm_file%pioFileDesc)
+    call errorHandle("PIO ERROR: issue arose with PIO_enddef for file"//trim(current_atm_file%filename),ierr)
+
+  end subroutine eam_init_pio_2
+!=====================================================================!
+  subroutine register_dimension(pio_atm_filename,shortname,longname,length)
+    character(len=*), intent(in)        :: pio_atm_filename
+    character(len=*), intent(in)        :: shortname,longname
+    integer, intent(in)                 :: length
+
+    type(pio_atm_output), pointer       :: pio_atm_file
+    type(hist_coord_t), pointer         :: hist_coord
+    type(hist_coord_list), pointer      :: curr=>null(), prev=>null()
+    integer :: ierr
+    character(len=150) :: errstr
+    logical :: found
+ 
+    call get_pio_atm_File(trim(pio_atm_filename),pio_atm_file,found)
+    if (.not.found) call errorHandle("PIO Error: Could not register variable"//trim(shortname)//", pio file "//trim(pio_atm_filename)//" not found",-999)
+    ! Get a new variable pointer in var_list
+    curr => pio_atm_file%coord_list_top
+    do while (associated(curr))
+      prev => curr
+      curr => prev%next
+    end do
+    allocate(prev%next)
+    curr => prev%next
+    allocate(curr%coord)
+    hist_coord => curr%coord
+    pio_atm_file%dimcounter = pio_atm_file%dimcounter + 1
+    ! Register this dimension
+    hist_coord%name      = trim(shortname)
+    hist_coord%long_name = trim(longname)
+    hist_coord%dimsize   = length
+    if (length.eq.0) then
+      ierr = PIO_def_dim(pio_atm_file%pioFileDesc, trim(shortname), pio_unlimited , hist_coord%dimid)
+    else
+      ierr = PIO_def_dim(pio_atm_file%pioFileDesc, trim(shortname), length , hist_coord%dimid)
+    end if
+    call errorHandle("PIO ERROR: could not define dimension "//trim(shortname)//" on file: "//trim(pio_atm_filename),ierr)
+    
+    return
+  end subroutine register_dimension
+!=====================================================================!
+  subroutine register_variable(pio_atm_filename,shortname,longname,length,dims,dtype,tag)
+    character(len=*), intent(in) :: pio_atm_filename
+    character(len=*), intent(in) :: shortname,longname
+    integer, intent(in)          :: length
+    character(len=*), intent(in) :: dims(length)
+    integer, intent(in)          :: dtype
+    character(len=*), intent(in) :: tag     ! Unique tag for this variables decomposition type
```

Comment:
What's the difference between the tag and the long/short name? It seems to me that also the long name has to be unique. Is the tag a short-short-name? Is the short name not unique?

Reply:
This is a construct of typical EAM output.  The shortname is unique and the long name doesn't have to be.  The shortname will be the variable name in the netCDF output.  The longname will accompany the shortname in the netCDF metadata as a descriptor.  Something like shortname=T, longname="atmospheric temperature in degrees K".  

Moving past the shortname/longname question.  The "tag" variable is a poor choice of variable name.  I will change it.  The "tag" is the unique "tag" associated with the pio-decomposition - which is associated with the grid it will be written on.  In other words, it is a way to check whether a new variable can use the pio-decomposition from a previously defined variable.

Reply:
Ah, gotcha. So yeah, something like `pio_decomp_tag` is probably better.